thrust/thrust/detail/contiguous_storage.inl

```diff
@@ -193,13 +194,17 @@ _CCCL_HOST_DEVICE void contiguous_storage<T, Alloc>::deallocate() noexcept
 template <typename T, typename Alloc>
 _CCCL_HOST_DEVICE void contiguous_storage<T, Alloc>::swap(contiguous_storage& x)
 {
-  thrust::swap(m_begin, x.m_begin);
-  thrust::swap(m_size, x.m_size);
+  using ::cuda::std::swap;
+  swap(m_begin, x.m_begin);
+  swap(m_size, x.m_size);
 
+  // FIXME(bgruber): swap_allocators already swaps m_allocator, so we are swapping twice here !!
   swap_allocators(integral_constant<bool, allocator_traits<Alloc>::propagate_on_container_swap::value>(),
                   x.m_allocator);
-
-  thrust::swap(m_allocator, x.m_allocator);
+  // FIXME(bgruber): this should use ADL-two-step swap, but this creates an ambiguity with std::swap until
+  // https://github.com/NVIDIA/cccl/issues/2984 is resolved.
+  // swap(m_allocator, x.m_allocator);
+  ::cuda::std::swap(m_allocator, x.m_allocator);
```

Comment:
So this looks like a bug to me. `swap_allocators` was introduced in `403effbd86aeb4dc1eccb135324f2ac0cc353d3d` (authored by @griwes), but did not replace swapping the allocator below. @griwes, I assume the `::cuda::std::swap(m_allocator, x.m_allocator);` is not needed/wrong here?

Reply:
The code is actually correct, because we are always swapping the allocator and swap it back if that wasn't ok.